src/opening-period/form/OpeningPeriodForm.tsx

```diff
@@ -0,0 +1,315 @@
+      start_time: `${timeSpan.startTime}:00`,
+      resource_state: timeSpan.resourceState,
+      weekdays: timeSpan.weekdays.reduce(
+        (acc: Array<number>, currentValue: boolean, currentIndex: number) => {
+          if (currentValue) {
+            acc.push(currentIndex + 1);
+            return acc;
+          }
+          return acc;
+        },
+        []
+      ),
+    };
+  });
+}
+
+const dropMilliseconds = (time: string): string => time.slice(0, -3);
+const getLowestWeekdayNumber = (timeSpan: TimeSpanApiFormat): number =>
+  timeSpan && timeSpan.weekdays ? timeSpan.weekdays.sort()[0] : 0;
+
+function formatApiTimeSpansToFormFormat(
+  apiTimeSpans: TimeSpanApiFormat[]
+): TimeSpanFormFormat[] {
+  return apiTimeSpans
+    .sort((a: TimeSpanApiFormat, b: TimeSpanApiFormat) => {
+      return getLowestWeekdayNumber(a) - getLowestWeekdayNumber(b);
+    })
+    .map((apiTimeSpan) => {
+      const weekdays: boolean[] = Array(7)
+        .fill(false)
+        .map((current, index): boolean =>
+          apiTimeSpan.weekdays
+            ? apiTimeSpan.weekdays.includes(index + 1)
+            : false
+        );
+
+      return {
+        id: apiTimeSpan.id,
+        description: apiTimeSpan.description?.fi || '',
+        startTime: apiTimeSpan.start_time
+          ? dropMilliseconds(apiTimeSpan.start_time)
+          : '',
+        endTime: apiTimeSpan.end_time
+          ? dropMilliseconds(apiTimeSpan.end_time)
+          : '',
+        resourceState: apiTimeSpan.resource_state || ResourceState.OPEN,
+        weekdays: weekdays as FormWeekdays,
+      };
+    });
+}
+
+const formatResourcePeriodOptionsToFormFormat = (
+  resourceStateApiOptions: ResourceStateApiOption[]
+): ResourceStateOption[] =>
+  resourceStateApiOptions.map((optionInApiFormat: ResourceStateApiOption): {
+    value: string;
+    label: string;
+  } => {
+    return {
+      value: optionInApiFormat.value,
+      label: `${
+        typeof optionInApiFormat.display_name === 'string'
+          ? optionInApiFormat.display_name
+          : optionInApiFormat.display_name.fi
+      }`,
+    };
+  });
+
+export default function OpeningPeriodForm({
+  formId,
+  datePeriod,
+  resourceId,
+  datePeriodOptions,
+  submitFn,
+  successTextAndLabel,
+  errorTextAndLabel,
+}: {
+  formId: string;
+  datePeriod?: DatePeriod;
+  resourceId: number;
+  datePeriodOptions: DatePeriodOptions;
+  submitFn: (datePeriod: DatePeriod) => Promise<DatePeriod>;
+  successTextAndLabel: NotificationTexts;
+  errorTextAndLabel: NotificationTexts;
+}): JSX.Element {
+  const language = Language.FI;
+
+  const resourceStateOptionsInApiFormat: ResourceStateApiOption[] =
+    datePeriodOptions.actions.POST.resource_state.choices;
+
+  const resourceStateOptions: ResourceStateOption[] = formatResourcePeriodOptionsToFormFormat(
+    resourceStateOptionsInApiFormat
+  );
+
+  const spans: TimeSpanFormFormat[] | {}[] =
+    datePeriod?.time_span_groups[0] &&
```

Comment:
I wonder if we could/should show some kind of warning in the UI if API returns date period with more than one time span group? Though since we are adding the groups to the UI soon, it would be temporary and probably not worth it then.